src/extension.ts

```diff
@@ -28,15 +28,29 @@ import { loadServerDefaults } from './common/setup';
 import { getLSClientTraceLevel, getProjectRoot } from './common/utilities';
 import { createOutputChannel, onDidChangeConfiguration, registerCommand } from './common/vscodeapi';
 import KedroVizPanel from './webview/vizWebView';
+import { installPythonDependencies } from './common/installPythonDependencies';
 
 let lsClient: LanguageClient | undefined;
 
 export async function getlsClient() {
     return lsClient;
 }
 
-
 export async function activate(context: vscode.ExtensionContext): Promise<void> {
+    const alreadyInstalled = context.globalState.get('dependenciesInstalled', false);
```

Comment:
What is this globalState? Does that mean it only install if the dependency is not there? I couldn't find the code where this state is set.

Reply:
In a VSCode extension, `globalState` is an API provided by the VSCode framework that allows you to store and retrieve global state info. This state is persisted across VS Code sessions, it remains available even after VS Code is closed and reopened. 

`globalState` does not have `.set` method.
<img width="739" alt="Screenshot 2024-08-28 at 6 30 37 p m" src="https://github.com/user-attachments/assets/4b5f20c9-a02a-4bdb-923d-d7e9caf0b2c3">

> Does that mean it only install if the dependency is not there? 

No it just check if `await installPythonDependencies(context);` is called before with `alreadyInstalled` flag.